graphql_compiler/schema_transformation/merge_schemas.py

```diff
@@ -338,8 +375,12 @@ def _process_scalar_definition(scalar, existing_scalars, type_name_to_schema_id,
             )
         )
     # new, valid scalar
-    merged_schema_ast.definitions.append(scalar)
-    existing_scalars.add(scalar_name)
+    new_definitions = list(merged_schema_ast.definitions)
+    new_definitions.append(scalar)
+    new_merged_schema_ast = ast_types.DocumentNode(definitions=new_definitions)
+    new_existing_scalars = deepcopy(existing_scalars)
```

Comment:
Similarly, this should be able to be just a `set()`/`copy()` call instead of a deep copy.